typedspark/_schema/get_schema_definition.py

```diff
@@ -64,6 +66,25 @@ def _build_schema_definition_string(
     return lines
 
 
+def _replace_literal_by_intervaltype(typehint, literal, intervaltype):
+    """Replace a Literal by a IntervalType, e.g. Literal[0] by IntervalType.DAY."""
+    return re.sub(
+        r"DayTimeIntervalType\[[^]]*\]",
+        lambda x: x.group(0).replace(literal, intervaltype),
+        typehint,
+    )
+
+
+def _replace_literals_in_day_time_interval_type(typehint):
+    """Replace all Literals in a DayTimeIntervalType by IntervalTypes, e.g. Literal[0]
+    by IntervalType.DAY."""
+    typehint = _replace_literal_by_intervaltype(typehint, "Literal[0]", "IntervalType.DAY")
```

Comment:
Is there some way to make the mapping from the Literal to the intervaltype defined in a dict and then just loop through the dict instead of hardcoding the mapping into 4 function calls?

Reply:
Good point, done!